colour/volume/macadam_limits.py

```diff
@@ -108,3 +109,186 @@ def is_within_macadam_limits(xyY, illuminant, tolerance=None):
+    except AttributeError:
+        illuminant = SDS_ILLUMINANTS['E']
+
+    # If there is no illuminant or it has the wrong form,
+    # an illuminant chosen with no influence
+    # If the illuminants do not match the format of the Standard Observer,
+    # they have to be adapted
+    illuminant.extrapolate(SpectralShape(360, 830))
+    illuminant.interpolate(SpectralShape(360, 830, 1))
+    # The cie31 cmfs are convolved with the given illuminant
+    X_illuminated = X_cie31 * illuminant.values
+    Y_illuminated = Y_cie31 * illuminant.values
+    Z_illuminated = Z_cie31 * illuminant.values
+    # Generate empty output-array
+    out_limits = np.zeros_like(standard_cfms.values)
+    # This Array has 471 entries for wavelengths from 360 nm to 830 nm
+    opti_colour = np.zeros_like(Y_illuminated)
+    # The array of optimal colours has the same dimensions like Y_illuminated
+    # and all entries are initially set to zero
+    middle_opti_colour = 235
+    # is a constant and not be changed. At 595nm (360 + 235)
+    # in the middle of the center_opti_colour-array
+    # be aware that counting in array-positions starts at zero
+    # The first optimum color has its center initially at zero
+    maximum_brightness = np.sum(Y_illuminated)
+
+    # "integral" over Y_illuminated
+
+    def optimum_colour(width, center):
+        opti_colour = np.zeros(471)
+        # creates array of 471 zeros and ones which represents optimum-colours
+        # All values of the opti_colour-array are initially set to zero
+        half_width = width
+        center_opti_colour = center
+        middle_opti_colour = 235
+        opti_colour[middle_opti_colour - half_width:middle_opti_colour +
+                    half_width + 1] = 1
+        # we start the construction of the optimum color
+        # at the center of the opti_colour-array
+        opti_colour = np.roll(opti_colour,
+                              center_opti_colour - middle_opti_colour)
+        # the optimum colour is rolled to the right wavelength
+        return opti_colour
+
+    def bright_opti_colour(width, center, lightsource):
+        brightness = np.sum(
+            optimum_colour(width, center) * lightsource) / maximum_brightness
+        return brightness
+
+    step_size = np.array([64, 32, 16, 8, 4, 2, 1])
+    for wavelength in range(0, 471):
+        width = 127
+        for n in step_size:
+            brightness = bright_opti_colour(width, wavelength, Y_illuminated)
+            if brightness > target_bright or width > 234:
+                width -= n
+            else:
+                width += n
+
+        brightness = bright_opti_colour(width, wavelength, Y_illuminated)
+        if brightness < target_bright:
+            width += 1
+            brightness = bright_opti_colour(width, wavelength, Y_illuminated)
+
+        rough_optimum = optimum_colour(width, wavelength)
+        brightness = np.sum(rough_optimum * Y_illuminated) / maximum_brightness
```

Comment:
```suggestion
        brightness = bright_opti_colour(width, wavelength, Y_illuminated)
        if brightness < target_bright:
            width += 1

        rough_optimum = optimum_colour(width, wavelength)
        brightness = np.sum(rough_optimum * Y_illuminated) / maximum_brightness
```

Seems like we can get rid of a brightness computation

Reply:
You are right. I tried it out, and it worked perfectly. Thank You again.